data-generator.tpl.ts

```diff
@@ -84,13 +92,59 @@ const createFoodProductImageValue = {
   },
 };
 
+const rawImageUrlToDownloadUrl = (imageUrl: string) => {
+  const width = 200;
+  const height = width * 2;
+  return `${imageUrl}?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=${width}&h=${height}&q=80&fm=jpg`;
+};
+
 const createBookImageUrlValue = {
   createValue: (seed: number, _refTime: number) => {
-    const width = 200;
-    const height = width * 2;
     const imageBackgroundUrl = random(options['BookImageBackground.src'], seed);
+    return rawImageUrlToDownloadUrl(imageBackgroundUrl);
+  },
+};
+
+const waitForImageFetchingToFinish = () => {
+  return new Promise<void>((resolve) => {
+    (function waitForFlagChange() {
+      if (!imageFetchingInProgress) return resolve();
+      setTimeout(waitForFlagChange, 50);
+    })();
+  });
+};
+
+const ensureImageData = async () => {
+  if (imageFetchingInProgress) {
+    await waitForImageFetchingToFinish();
+    return;
+  }
+  if (imageDataList.length > 0) {
+    return;
+  }
+  imageFetchingInProgress = true;
+
+  const imageUrls = options['BookImageBackground.src'];
+  if (!imageUrls) {
+    throw new Error('No imageUrls');
+  }
+  const promises: Promise<void>[] = [];
+
+  for (const imageUrl of imageUrls) {
+    promises.push(
+      fetch(rawImageUrlToDownloadUrl(imageUrl)).then(async (response) => {
+        imageDataList.push(await response.arrayBuffer());
+      })
+    );
+  }
+  await Promise.all(promises);
+  imageFetchingInProgress = false;
+};
 
-    return `${imageBackgroundUrl}?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=${width}&h=${height}&q=80&fm=jpg`;
+const createBookImageDataValue = {
+  createValue: async (seed: number, _refTime: number) => {
+    await ensureImageData();
```

Comment:
As discussed, this can be simplified to
1. pick a url
2. fetch that url
3. return the array buffer for it

The browser should take care of all needed caching

Reply:
Done. Seems to work in Start as you expected.